state.go

```diff
@@ -317,6 +292,66 @@ func (m *Memberlist) probeNode(node *nodeState) {
 	m.suspectNode(&s)
 }
 
+// Ping initiates a ping to the node with the specified name.
+func (m *Memberlist) Ping(node string) (bool, time.Duration, error) {
```

Comment:
Yep, that makes sense.  I was thinking you could return a specific error in the retry case and check for that error in the caller to determine whether it's transient or not.  Renaming the return value I think would be helpful in understanding the intent.